parts/k8s/windowskubeletfunc.ps1

```diff
@@ -605,7 +664,161 @@ catch
+
+
+    `$configJson.AdditionalArgs[0].Value.Settings.Exceptions[0] = `$global:KubeClusterCIDR
+    `$configJson.AdditionalArgs[0].Value.Settings.Exceptions[1] = `$global:MasterSubnet
+    `$configJson.AdditionalArgs[1].Value.Settings.DestinationPrefix  = `$global:KubeServiceCIDR
+
+    if (Test-Path `$global:CNIConfig)
+    {
+        Clear-Content -Path `$global:CNIConfig
+    }
+
+    Write-Host "Generated CNI Config [`$configJson]"
+
+    Add-Content -Path `$global:CNIConfig -Value (ConvertTo-Json `$configJson -Depth 20)
+}
+
+try
+{
+    `$masterSubnetGW = Get-DefaultGateway `$global:MasterSubnet
+    `$podCIDR=Get-PodCIDR
+    `$podCidrDiscovered=Test-PodCIDR(`$podCIDR)
+
+    # if the podCIDR has not yet been assigned to this node, start the kubelet process to get the podCIDR, and then promptly kill it.
+    if (-not `$podCidrDiscovered)
+    {
+        `$argList = $KubeletArgListStr
+
+        `$process = Start-Process -FilePath c:\k\kubelet.exe -PassThru -ArgumentList `$argList
+
+        # run kubelet until podCidr is discovered
+        Write-Host "waiting to discover pod CIDR"
+        while (-not `$podCidrDiscovered)
+        {
+            Write-Host "Sleeping for 10s, and then waiting to discover pod CIDR"
+            Start-Sleep 10
+
+            `$podCIDR=Get-PodCIDR
+            `$podCidrDiscovered=Test-PodCIDR(`$podCIDR)
+        }
+
+        # stop the kubelet process now that we have our CIDR, discard the process output
+        `$process | Stop-Process | Out-Null
+    }
+
+    # Turn off Firewall to enable pods to talk to service endpoints. (Kubelet should eventually do this)
+    netsh advfirewall set allprofiles state off
+
+    # startup the service
+    `$hnsNetwork = Get-HnsNetwork | ? Name -EQ `$global:NetworkMode.ToLower()
+
+    if (`$hnsNetwork)
+    {
+        # Kubelet has been restarted with existing network.
+        # Cleanup all containers
+        # TODO: convert this to ctr.exe -n k8s.io container list ; container rm
+        docker ps -q | foreach {docker rm `$_ -f}
+        # cleanup network
+        Write-Host "Cleaning up old HNS network found"
+        Remove-HnsNetwork `$hnsNetwork
+        Start-Sleep 10
+    }
+
+    Write-Host "Creating a new hns Network"
+    ipmo `$global:HNSModule
+
+    `$hnsNetwork = New-HNSNetwork -Type `$global:NetworkMode -AddressPrefix `$podCIDR -Gateway `$masterSubnetGW -Name `$global:NetworkMode.ToLower() -Verbose
+    # New network has been created, Kubeproxy service has to be restarted
+    Restart-Service Kubeproxy
+
+    Start-Sleep 10
+    # Add route to all other POD networks
+    Write-Host "Updating CNI config"
+    Update-CNIConfig `$podCIDR `$masterSubnetGW
+
+    $KubeletCommandLine
+}
+catch
+{
+    Write-Error `$_
+}
+
+"@
+    } # end elseif using sdnbridge and containerd.
+    else
+    {
+        Write-Error "The combination of $NetworkPlugin and UseContainerD=$UseContainerD is not implemented"

```

Comment:
can you throw here so we can get an error in app insights?

Reply:
sure

Reply:
done